packages/@lwc/engine-core/src/framework/runtime-instrumentation.ts

```diff
@@ -0,0 +1,11 @@
+type InstrumentedWindow = (Window & typeof globalThis) & {
+    __lwc_instrument_cmp_def?: (def: any) => void;
+    __lwc_instrument_cmp_instance?: (instance: any, vm: any) => void;
+};
+
+const win = typeof window !== 'undefined' ? (window as InstrumentedWindow) : null;
+
+function noop() {}
+
+export const instrumentDef = win?.__lwc_instrument_cmp_def ?? noop;
+export const instrumentInstance = win?.__lwc_instrument_cmp_instance ?? noop;
```

Comment:
Can we also add a check for `process.env.NODE_ENV !== 'production'`? I'm assuming this is only needed to run in dev mode.

Reply:
We do want dev tools to function against production sites.